### PR TITLE
fix: status clear

### LINKED ISSUE #174: StatusCondition.clear() has weird behavior.
## Current Behavior
Using `clear()` on a `StatusCondition` can be used to apply a status condition if it is not present. It can also be used to give it a different non-volatile status condition entirely, in a way that doesn't make any sense.

**To Reproduce**
Code snippet to reproduce the problem (if applicable)
```go
s := StatusNone
s.clear(StatusBurn)
if s == StatusBurn {
    panic("should not happen")
}
```

```go
s := StatusNone
s.clear(StatusConfusion)
if s == StatusConfusion {
    panic("should not happen")
}
```

```go
s := StatusParalyze
s.clear(StatusFreeze)
if s == StatusBurn {
    panic("should not happen")
}
```

## Expected Behavior
`clear` should only xor status conditions that are already present.

**Additional context**

### PR DESCRIPTION
**Pull request by adapap**
Simple enough? Closes #174

### CODE REVIEW

**Review by dyc3 — CHANGES_REQUESTED** [review]
Doesn't have any tests. The solution is definitely not that simple.

File: types_test.go [hunk 1]
```diff
@@ -239,6 +239,30 @@ func TestStatusConditionClear(t *testing.T) {
...
+		{
+			name:  "Should not clear status that does not exist",
+			from:  StatusNone,
+			clear: StatusBurn,
+			want:  StatusNone,
+		},
+		{
+			name:  "Should not clear freeze",
+			from:  StatusParalyze,
+			clear: StatusConfusion,
```
Comment (dyc3) [inline types_test.go:251]:
```suggestion
			clear: StatusFreeze,
```

**Review by dyc3 — CHANGES_REQUESTED** [review]

**Review by dyc3 — CHANGES_REQUESTED** [review]
I fixed the test case for you. The solution still does not fix the bug.

**Comment by adapap** [comment]
I decided to simplify the logic instead of making the status condition check more complex. It's a lot more maintainable this way IMO.

**Review by dyc3 — APPROVED** [review]

**Pull request merged**